src/fn/upload-app-settings.js

```diff
@@ -1,21 +1,95 @@
 const fs = require('../lib/sync-fs');
 const request = require('request-promise-native');
 const skipFn = require('../lib/skip-fn');
+const inquirer = require('inquirer');
+const jsonDiff = require('json-diff');
+const fse = require('fs-extra');
+
+const actionChoices = [
+  { name: 'Overwrite the changes', value: 'overwrite' }, 
+  { name: 'Abort so that you can update the configuration', value: 'abort' }
+];
+
+const actionQuestionsWithDiff = [{
+  type: 'list',
+  name: 'action',
+  message: 'You are trying to modify a configuration that has been modified since your last upload. Do you want to?',
+  choices: actionChoices.concat([{ name: 'View diff', value: 'diff' }])
+}];
+
+const actionQuestionsWithoutDiff = [{
+  type: 'list',
+  name: 'action',
+  message: 'You are trying to modify a configuration that has been modified since your last upload. Do you want to?',
+  choices: actionChoices
+}];
+
+module.exports = async (projectDir, couchUrl) => {
+  const body = fs.read(`${projectDir}/app_settings.json`);
 
-module.exports = (projectDir, couchUrl) => {
   if(!couchUrl) return skipFn('no couch URL set');
 
-  return request
-    .put({
-      method: 'PUT',
-      url: `${couchUrl}/_design/medic/_rewrite/update_settings/medic?replace=1`,
-      headers: { 'Content-Type':'application/json' },
-      body: fs.read(`${projectDir}/app_settings.json`),
-    })
-    .then(JSON.parse)
-    .then(json => {
-      // As per https://github.com/medic/medic-webapp/issues/3674, this endpoint
-      // will return 200 even when upload fails.
-      if(!json.success) throw new Error(json.error);
+  // Pull remote _rev
+  let remoteRev = 'remoteRev';
+  let remoteJson = '';
+  try {
+    const response = await request.get({
+      method: 'GET',
+      url: `${couchUrl}/settings`,
+    });
+    remoteJson = JSON.parse(response);
+    remoteRev = remoteJson._rev;
+  } catch (e) {
+    // continue regardless of error
+  }
+
+  // Pull local _rev
+  let localRev = 'localRev';
+  try {
+    localRev = fs.read(`${projectDir}/._revs/settings`);
+  } catch (e) {
+    // continue regardless of error
```

Comment:
I wonder if we need a different option when there isn't any localRev, ie: the first time someone upgrades to this version they will get prompted. Instead we could say something like "we can't determine if you're going to overwrite someone else's changes or not. Would you like to proceed?"